exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -202,6 +203,20 @@ object SchemaUtils {
                                 is MysqlDialect -> "_utf8mb4\\'${processed.trim('(', ')', '\'')}\\"
                                 else -> processed.trim('\'')
                             }
+                        } else if (column.columnType is ArrayColumnType && dialect is PostgreSQLDialect) {
```

Comment:
Please consider using when statement. We can replace `when (exp) {` -> `when {` and flatten many conditions and  extract some to functions with readable names